src/Pinch/Generate.hs

```diff
@@ -159,10 +159,39 @@ gInclude dir i = do
 
 gDefinition :: Definition SourcePos -> GenerateM ([H.Decl], [H.Decl], [H.Decl])
 gDefinition def = case def of
-  ConstDefinition _ -> pure ([], [], [])
+  ConstDefinition c -> (\x -> (x, [], [])) <$> gConst c
   TypeDefinition ty -> (\x -> (x, [], [])) <$> gType ty
   ServiceDefinition s -> gService s
 
+gConst :: A.Const SourcePos -> GenerateM [H.Decl]
+gConst const = do
+  tyRef <- gTypeReference (constValueType const)
+  value <- gConstValue (constValue const)
+  pure
+    [ H.TypeSigDecl name tyRef
+    , H.ConstDecl name value
+    ]
+  where
+    name = decapitalize (constName const)
+
+gConstValue :: A.ConstValue SourcePos -> GenerateM H.Lit
+gConstValue val = case val of
+  ConstInt n _ -> pure (H.LInt n)
+  ConstFloat n _ -> pure (H.LFloat n)
+  ConstLiteral s _ -> pure (H.LString s)
+  ConstIdentifier id _
+    | xs @(_:_:_) <- T.splitOn "." id -> do
+      map <- asks cModuleMap
+      case Map.lookup (mconcat $ init xs) map of
+        Nothing ->
+          -- TODO this should probably be an error
+          pure (H.LRef Nothing (decapitalize id))
+        Just mod ->
+          pure $ H.LRef (Just mod) (decapitalize (last xs))
+    | otherwise -> pure $ H.LRef Nothing (decapitalize id)
+  ConstList xs _ -> H.LList <$> traverse gConstValue xs
```

Comment:
Hm, I think doing this without overloaded lists would be `H.EApp "Data.Vector.fromList" (H.EList ....)`. I think that would not be much more onerous and doesn't need the overloaded lists extension. I would prefer that.